test/OpenTelemetry.Tests/Logs/LogRecordTest.cs

```diff
@@ -107,6 +107,61 @@ public void CheckStateForUnstructuredLogWithStringInterpolation()
             Assert.Equal(message.ToString(), state.ToString());
         }
 
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1602:Enumeration items should be documented", Justification = "enum for test")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1201:Elements should appear in the correct order", Justification = "enum for test")]
+        public enum Field
+        {
+            FormattedMessage,
+            State,
+            StateValues,
+        }
+
+        [Theory]
+        [InlineData(Field.State)]
+        [InlineData(Field.StateValues)]
+        [InlineData(Field.FormattedMessage)]
+        public void SettersTest(Field fieldName)
+        {
+            var exportedItems = new List<LogRecord>();
+            var exporterLoggerFactory = LoggerFactory.Create(builder => builder
```

Comment:
just my thought: Suggesting to make this into 3 separate tests to make it easier to read.

Reply:
For example, the FormattedMessageCanBeSetByProcessor, StateCanBeSetByProcessor etc.

Reply:
I've updated as suggested. Thanks!